src/API.jl

```diff
@@ -1008,7 +1009,7 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
     end
     depsmap = Dict{Base.PkgId, Vector{Base.PkgId}}(Iterators.filter(!isnothing, deps_pair_or_nothing)) #flat map of each dep and its deps
 
-    if ctx.env.pkg !== nothing && isfile( joinpath( dirname(ctx.env.project_file), "src", ctx.env.pkg.name * ".jl") )
+    if ctx.env.pkg !== nothing && isfile( joinpath( dirname(ctx.env.project_file), "src", string(ctx.env.pkg.name, ".jl")) )
```

Comment:
I think the standard way of doing this is:

```
ctx_env_pkg = ctx.env.pkg
if ctx_env_pkg !== nothing && isfile( joinpath( dirname(ctx.env.project_file), "src", ctx_env_pkg.name * ".jl") )
```

Basically, hoist the load of the thing we check for `nothing` so that Julia realizes it cannot have changed the next time we use it. Changing `*` to `string` is mostly a bandaid.